examples/spotify-api-example/src/main/java/com/spotify/apollo/example/AlbumResource.java

```diff
@@ -0,0 +1,109 @@
+package com.spotify.apollo.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spotify.apollo.Client;
+import com.spotify.apollo.Request;
+import com.spotify.apollo.RequestContext;
+import com.spotify.apollo.Response;
+import com.spotify.apollo.Status;
+import com.spotify.apollo.example.data.Album;
+import com.spotify.apollo.example.data.Artist;
+import com.spotify.apollo.route.AsyncHandler;
+import com.spotify.apollo.route.Middlewares;
+import com.spotify.apollo.route.Route;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.StringJoiner;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+
+import okio.ByteString;
+
+/**
+ * The album resource demonstrates how to use asynchronous routes in Apollo.
+ */
+public class AlbumResource {
+
+  private static final String SPOTIFY_API = "https://api.spotify.com";
+  private static final String SEARCH_API = SPOTIFY_API + "/v1/search";
+  private static final String ALBUM_API = SPOTIFY_API + "/v1/albums";
+
+  public Stream<Route<AsyncHandler<Response<ByteString>>>> routes() {
+    // The album resource has two routes. Both are using the autoserializer.
+    return Stream.of(
+        Route.async("GET", "/albums/new", this::getAlbums),
+        Route.async("GET", "/albums/hipster", this::getAlbums)
+    )
+        .map(route -> route.withMiddleware(Middlewares::autoSerialize));
```

Comment:
I sort of wonder if we should use auto-serialisation in examples. Somebody said that the code you should really make absolutely perfect is example code, because it lives forever. And if we think (like I do) that it's better to be type-safe, then we should use type-safe serialisation.